x-pack/plugins/spaces/public/management/edit_space/roles/component/space_assign_role_privilege_form.tsx

```diff
@@ -0,0 +1,505 @@
+            color="primary"
+            idSelected={roleSpacePrivilege}
+            onChange={(id) => onRoleSpacePrivilegeChange(id as KibanaRolePrivilege)}
+            buttonSize="compressed"
+            isFullWidth
+          />
+        </EuiFormRow>
+        {roleSpacePrivilege === 'custom' && (
+          <EuiFormRow
+            data-test-subj="rolePrivilegeCustomizationForm"
+            label={i18n.translate(
+              'xpack.spaces.management.spaceDetails.roles.assign.privileges.customizeLabelText',
+              { defaultMessage: 'Customize by feature' }
+            )}
+          >
+            <>
+              <EuiText size="xs">
+                <p>
+                  <FormattedMessage
+                    id="xpack.spaces.management.spaceDetails.roles.assign.privileges.customizeDescriptionText"
+                    defaultMessage="Increase privilege levels per feature basis. Some features might be hidden by the
+                    space or affected by a global space privilege"
+                  />
+                </p>
+              </EuiText>
+              <EuiSpacer />
+              <React.Fragment>
+                {!kibanaPrivileges ? (
+                  <EuiLoadingSpinner size="l" />
+                ) : (
+                  <KibanaPrivilegeTable
+                    role={roleCustomizationAnchor.value!}
+                    privilegeIndex={roleCustomizationAnchor.privilegeIndex}
+                    onChange={(featureId, selectedPrivileges) => {
+                      // apply selected changes only to customization anchor, this way we delay reconciling the intending privileges
+                      // of the selected roles till we decide to commit the changes chosen
+                      setRoleCustomizationAnchor(({ value, privilegeIndex }) => {
+                        let privilege;
+
+                        if ((privilege = value!.kibana?.[privilegeIndex!])) {
+                          privilege.feature[featureId] = selectedPrivileges;
+                        }
+
+                        return { value, privilegeIndex };
+                      });
+                    }}
+                    onChangeAll={(_privilege) => {
+                      // dummy function we wouldn't be using this
+                    }}
+                    kibanaPrivileges={new KibanaPrivileges(kibanaPrivileges, features)}
+                    privilegeCalculator={
+                      new PrivilegeFormCalculator(
+                        new KibanaPrivileges(kibanaPrivileges, features),
+                        roleCustomizationAnchor.value!
+                      )
+                    }
+                    allSpacesSelected={false}
+                    canCustomizeSubFeaturePrivileges={false}
+                  />
+                )}
+              </React.Fragment>
+            </>
+          </EuiFormRow>
+        )}
+      </EuiForm>
+    );
+  };
+
+  const getSaveButton = () => {
+    return (
+      <EuiButton
```

Comment:
__question:__ is it intentional that this button is enabled even if none new roles are selected?